operator/pkg/apis/cassandra/v1beta1/webhook.go

```diff
@@ -65,18 +75,33 @@ func ValidateDatacenterFieldChanges(oldDc CassandraDatacenter, newDc CassandraDa
 	newRacks := newDc.GetRacks()
 
 	if len(oldRacks) > len(newRacks) {
-		return fmt.Errorf("CassandraDatacenter attempted to remove Rack")
+		return attemptedTo("remove Rack")
+	}
+
+	newRackCount := len(newRacks) - len(oldRacks)
+	if newRackCount > 0 {
+		newSizeDifference := newDc.Spec.Size - oldDc.Spec.Size
+
+		if newSizeDifference <= 0 {
+			return attemptedTo("add Rack without increasing Size")
+		}
+
+		if int(newSizeDifference) < newRackCount {
+			return attemptedTo(
+				fmt.Sprintf("add Racks without increasing Size enough to populate each rack.\n" +
+					"New racks added: %d, Size increased by: %d", newRackCount, newSizeDifference))
+		}
```

Comment:
The line `newRackCount := len(newRacks) - len(oldRacks)` doesn't look right with the check `if int(newSizeDifference) < newRackCount`. How would this ever work if I added a single rack? Ideally, we'd leverage the same logic used in reconciliation to determine rack sizes from node counts so that validation stays aligned with reconciliation.

Reply:
Err... wait, I misread this... still not sure if the logic is right here though. If I go from 9 nodes on 2 racks to 11 nodes on 3 racks, won't I get the following:

```
rack 1 -> 5 nodes
rack 2 -> 4 nodes
```

to

```
rack 1 -> 4 nodes
rack 2 -> 4 nodes
rack 3 -> 3 nodes
```

Causing rack 1 to have fewer nodes?

Reply:
yeah after thinking about this for too many hours, I think a simple way to phrase the solution is "you have to add as many nodes as the smallest rack had before the increase *times* the number of racks you're adding"

Reply:
My horrible ASCII art logic

```
9 nodes in 2 racks
xxxxx
xxxx

13 nodes in 3 racks
xxxxx
xxxx
xxxx
```